src/library/pia/pia-format.ts

```diff
@@ -569,8 +569,115 @@ Maximum wage base projection indicator
+        .setEndChar(9),
+    };
+  }
+  serialize(data: Partial<PiaTypes.PiaData>): PiaTypes.PiaLineMap {
+    const line8 = {
+      8: `08${
+        data.forwardProjectionType != undefined
+          ? data.forwardProjectionType
+          : this.fieldFormats.forwardProjectionType.getBlank()
+      }${
+        data.forwardProjectionPercentage != undefined
+          ? data.forwardProjectionPercentage //TODO handle string conversion
+          : this.fieldFormats.forwardProjectionPercentage.getBlank()
+      }${
+        data.lastYearForwardEarningsProjections != undefined
+          ? data.lastYearForwardEarningsProjections
+          : this.fieldFormats.lastYearForwardEarningsProjections.getBlank()
+      }`,
+    };
+    const line40 = {
+      40: `40${
+        data.firstYearBenefitProjection != undefined
+          ? data.firstYearBenefitProjection
+          : this.fieldFormats.firstYearBenefitProjection.getBlank()
+      }
+      ${
+        data.benefitIncreaseAssumption != undefined
+          ? data.benefitIncreaseAssumption
+          : this.fieldFormats.benefitIncreaseAssumption.getBlank()
+      }${
+        data.avgWageIncreaseAssumption != undefined
+          ? data.avgWageIncreaseAssumption
+          : this.fieldFormats.avgWageIncreaseAssumption.getBlank()
+      }${
+        data.maxWageBaseProjectionInd != undefined
+          ? data.maxWageBaseProjectionInd
+          : this.fieldFormats.maxWageBaseProjectionInd.getBlank()
+      }`,
+    };
+    const hasAnyLine8 =
+      data.forwardProjectionType ||
+      data.forwardProjectionPercentage ||
+      data.lastYearForwardEarningsProjections;
+    const hasAnyLine40 =
+      data.firstYearBenefitProjection ||
+      data.benefitIncreaseAssumption ||
+      data.avgWageIncreaseAssumption ||
+      data.maxWageBaseProjectionInd;
+    return { ...(hasAnyLine8 && line8), ...(hasAnyLine40 && line40) };
+  }
+  deserialize(lineMap: PiaTypes.PiaLineMap): Partial<PiaTypes.PiaData> {
+    let line8 = lineMap[8];
+    let line8Data = line8
+      ? {
+          forwardProjectionType: parseInt(
+            PiaUtils.piaSubstr(
+              line8,
+              this.fieldFormats.forwardProjectionType.startChar,
+              this.fieldFormats.forwardProjectionType.endLine
+            ),
+            10
+          ),
+          forwardProjectionPercentage: parseInt(
+            PiaUtils.piaSubstr(
+              line8,
+              this.fieldFormats.forwardProjectionPercentage.startLine,
+              this.fieldFormats.forwardProjectionPercentage.endLine
+            ),
+            10
+          ),
+          lastYearForwardEarningsProjections: parseInt(
+            PiaUtils.piaSubstr(
+              line8,
+              this.fieldFormats.lastYearForwardEarningsProjections.startLine,
+              this.fieldFormats.lastYearForwardEarningsProjections.endChar
+            ),
+            10
+          ),
+        }
+      : {};
+    return { ...line8Data };
```

Comment:
TODO finish deserializer for line40data and the getter and setter. We can do this on `line8line40` branch which has the new tests for sample25.pia (which will still need some tweaking)